libcontainer/cgroups/devices/devicefilter_test.go

```diff
@@ -136,10 +217,6 @@ block-11:
         62: MovImm32 dst: r0 imm: 0
         63: Exit
 `
-	var devices []*devices.Rule
-	for _, device := range specconv.AllowedDevices {
-		devices = append(devices, &device.Rule)
-	}
 	testDeviceFilter(t, devices, expected)
```

Comment:
I'm not familiar wit this code yet, so take this as a question rather than an ask to change something.

But `testDeviceFilter()` calls `deviceFilter()` which is not trivial. If we change the allowed devices, shall we adapt this test to be meaningful again? It's a real question that I don't know the answer. If we do, or if it's not clear, maybe the dependency on specconv is fine (or moving the allowedDevices here), as both things seem simple and specconv is quite small too.

I understand the comment on specconv.AllowedDevices changing it is risky, but we just tried to change it, maybe we won't try again soon, but we might try again at some point in the future.

Reply:
Basically, the test checks that the EBPF code generated is as expected, using the default set of devices as input. If we change the specconv.AllowedDevices, we'll have two options:
 - keep testing the "old" default set as input;
 - update the test case code (both the input and the expected EBPF).

Either way is fine TBH.

I don't like the duplication of the data here, but this is for #4618 (i.e. we're preparing to move this code into a separate repo, github.com/opencontainers/cgroups), and having this to be dependent on libcontainers/specconv just for the sake of a test case is worse from my POV.

I have outlined all the other options in the commit descriptions -- please review and let me know if you like any of them better than the current approach.

Reply:
Yes, I had that context in mind and was reviewing commit by commit, so I checked the commit msg :)

My reason to ask was the reliability of the test if we split this out with a c&p and we change the list over time. I'll trust your judgment here, as I said I'm not familiar with this part of the code yet, and do what you think it is best.

If we stay with this option _and_ you think it would be better to keep the test case using the updated list, I'd add a comment in the runc code saying to update the test in that case. But if you think it's not better to keep the test case using the updated list, it's fine like this :)